### PR TITLE
feat: auto import of style files

### PR DESCRIPTION
**Pull request by develar**
Addition for #26.

Your plugin is awesome. 

1. I do not have luck to make https://github.com/anncwb/vite-plugin-style-import working.
2. No need to use yet another plugin if in any case the goal is the same — import on demand. Style it is integral part of component.
3. Your package has less dependencies, I don't need another set.

So, I decided to add property `stylePath` to `ComponentInfo`. 

Only Vue 3 is supported by intention.

Example for `element-plus`: 

```ts
      customComponentResolvers: [
        name => {
          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-ignore
          if (name.startsWith("El")) {
            const partialName = name[2].toLowerCase() + name.substring(3).replace(/[A-Z]/g, l => `-${l.toLowerCase()}`)
            return {
              path: `element-plus/es/el-${partialName}`,
              stylePath: `element-plus/packages/theme-chalk/src/${partialName}.scss`,
            }
          }
          return null
        },
```

Two important notes: official documentation of element plus is wrong and you should not follow it if you want to get smaller production build. See https://github.com/element-plus/element-plus/issues/1492 and https://github.com/element-plus/element-plus/issues/1477

1. Do not use `element-plus`. Use `element-plus/es` instead to make sure that tree shaking works. It is Vite responsibility to transform modern syntax to old one if needed. 
2. Do not import CSS files (`element-plus/lib/theme-chalk/*.css`). Instead, install `sass` and import SCSS (`element-plus/packages/theme-chalk/src/*.scss`). Issue is that CSS files are not intended for on-demand loading. For example, `autocomplete.css` contains styles also for `input`, and if you use `el-input` in addition to `el-autocomplete`, you will get styles twice. It is Vite responsibility to bundle CSS files.

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -94,6 +94,7 @@ Required<Options>,
 export interface ComponentInfo {
   name: string
   path: string
+  stylePath?: string
```
Comment (antfu) [inline src/types.ts:97]: Would like to make it an array so it can be more flexible for other situations.

**Comment by antfu** [comment]
I will do it now

**Comment by develar** [comment]
> I will do it now

Just pushed array support and test :)

File: src/transforms/vue3.ts [hunk 2]
```diff
@@ -44,3 +44,27 @@ export function Vue3Transformer(ctx: Context): Transformer {
     return transformed
   }
 }
+
+export function addImports(context: Context, component: ComponentInfo, result: Array<string>, varName: string): void {
```
Comment (develar) [inline src/transforms/vue3.ts:48]: Function is extracted to be able to write a unit test.

**Comment by antfu** [comment]
I have renamed it to `sideEffects` to be more general and allowing named imports. Also make it available for Vue 2. Can you have a look? If that works for you, I am heading to the merge. Thanks a lot for this!

**Comment by develar** [comment]
I see that `sideEffects` is not supported as a string. Is it by intention? Maybe better to not require to use wrap into `[]` without need? Ok, not a big deal. 

Thanks! Indeed, handling `sideEffects` in this is much better, because it is possible now to add not only default import. But in this light property name is not fully correct — `{ name: 'css', path: 'test2.css' }` it is not side effect import, right? So, instead of introducing additional property, maybe will be better just support returning `Array<ImportInfo>` from `ComponentResolver`?

instead of
`declare type ComponentResolveResult = string | ComponentInfo;` 

will be:
`string | ImportInfo | Array<ImportInfo>`

**Comment by antfu** [comment]
> support returning Array<ImportInfo> from ComponentResolver

That's what I implement originally, but later I find it's not ideal as I would need to replace the `__resolveComponent` with a single import. Treating the first one of the array as the "main" component to replace does not make much sense to me. So I think it's better to have the current parent-child relations to describe it.

`{ name: 'css', path: 'test2.css' }` is not necessary an side effect import (but possible). But on the other hand, maybe we can think about the "side effect" of importing a component, any additional changes could be considered as side-effects (polluting the naming scope, etc). Would this make sense to you?

> sideEffects is not supported as a string

Updated.

**Comment by develar** [comment]
Thanks for clarification. That's makes sense for me.

**Pull request merged**